pkg/controller/backup_finalizer_controller.go

```diff
@@ -102,6 +105,10 @@ func (r *backupFinalizerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	original := backup.DeepCopy()
 	defer func() {
+		switch original.Status.Phase {
```

Comment:
I think we need to check `backup.Status.Phase`? because this one is to be updated in the code following